woocommerce/woocommerce-ios
# [HACK week] Networking: Process network response in background thread

Closes: #11255


## Description

### What? 

The networking layer Mappers parses a networking response into models from the main thread.

### Why?

When a large network response is mapped from the main thread, the main thread gets blocked.

A blocked main thread results in app hangs and the watchdog can terminate the app if the main thread is blocked for a significant amount of time.

### How? 

Process the network response from the background thread.

**Changes**
1. Make `map` function async and update all mapper files.
2. Process network response in `Remote` inside a `Task`.
3. Update unit tests. 

⚠️ Sorry for the 500+ changes. I had to update all the mapper-related unit tests in the same PR. 

## Testing instructions

Smoke test the app using both WPCOM and wporg login and ensure that it works as before.

## Screenshots
NA

---
